### ISSUE TITLE
User Story: Password Reset

### ISSUE THREAD

**Issue by DasUnicorn** [DasUnicorn/WordWeave#18]
**Description Of Service Or Screen:**
Password Reset

**User Objective:**
As a WordWeave user, I want the ability to reset my password in case I forget it or need to enhance security.

**Acceptance Criteria:**

- Users can access the "Forgot Password" link on the login screen.
- Clicking on the link navigates the user to a password reset page.
- The password reset page includes a field for the user to enter their registered email address.
- Upon submission, the system checks the entered email against registered accounts and sends a password reset link to the provided email address.
- Users receive a confirmation message on the screen indicating that a password reset email has been sent.
- The password reset email includes a secure link with a time-limited token for security purposes.
- Clicking on the password reset link takes the user to a secure page where they can enter a new password.
- The password creation form enforces security requirements (e.g., minimum length, combination of letters and numbers).
- After successfully resetting the password, users receive a confirmation message, and they can log in using their updated credentials.

**Comment by DasUnicorn** [comment]
Functionality is provided by adding django AllAuth.
The profile page has to be finished first to implement this option.

**Comment by DasUnicorn** [comment]
**Testing**

| Testcase    | Result |
| -------- | ------- |
|Users can access the "Forgot Password" link on the login screen.  |yes    |
| Clicking on the link navigates the user to a password reset page.|yes |
| The password reset page includes a field for the user to enter their registered email address.  | new ticket: #27 , needs to be tested in production with email server etc    |
|The password reset email includes a secure link with a time-limited token for security purposes.    | New ticket: #27    |
|Clicking on the password reset link takes the user to a secure page where they can enter a new password.    | New ticket: #27    |
|The password creation form enforces security requirements (e.g., minimum length, combination of letters and numbers). | yes |
|After successfully resetting the password, users receive a confirmation message, and they can log in using their updated credentials. | yes |
